File: src/top_p_example.py
from openai import OpenAI

client = OpenAI()

prompt = "Write a short story about a robot learning to dance."

# Top P controls how much probability mass to sample from
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[{"role": "user", "content": prompt}],
    max_tokens=100,
    temperature=0.7,
    top_p=0.5   # nucleus sampling (only sample from tokens with cumulative prob ≤ 0.5)
)

print(response.choices[0].message["content"])

File: src/top_p_example.js
import OpenAI from "openai";

const client = new OpenAI();

const prompt = "Suggest three unique ice cream flavors.";

const run = async () => {
  for (const p of [0.3, 0.7, 1.0]) {
    const response = await client.chat.completions.create({
      model: "gpt-4o-mini",
      messages: [{ role: "user", content: prompt }],
      max_tokens: 50,
      top_p: p,
    });

    console.log(`\n--- top_p: ${p} ---`);
    console.log(response.choices[0].message.content);
  }
};

run();
